ddtrace/tracer/handler.go

```diff
@@ -0,0 +1,167 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+package tracer
+
+import (
+	"encoding/json"
+	"fmt"
+	"sync"
+	"time"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/log"
+)
+
+type traceHandler interface {
+	// pushPayload adds a trace to the current payload being constructed by the handler.
+	pushPayload([]*span)
+	// flush should cause the handler to send its current payload. The flush can happen asynchronously, but
+	// the handler must be ready to accept new traces with pushPayload when flush() returns.
+	flush()
```

Comment:
Is there a way from the public tracer API we can force a flush to occur? Will the flush happen as soon as soon as the root span completes, (since the process gets frozen almost instantly after that point).

Reply:
This is a good question. 

Since  the tracer needs to be started at each invocation, it also needs to be stopped at each invocation, and `flush` is forced when the tracer is stopped. I think that should work correctly as is.

Alternatively, we could flush every time a trace is finished. This could result in more log lines with fewer traces each.

Reply:
I think for the base case, there would only be one trace per invocation, since they would be using the lambda wrapper which creates the root span, https://github.com/DataDog/datadog-lambda-go .

Reply:
👍 
I think that will probably be the common case. However, it's quite easy to create additional, separate traces, and we don't want to assume there will be only one per invocation.